test/app-luatest/init_script_test.lua

```diff
@@ -0,0 +1,144 @@
+-- The test tests init script.
+
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local justrun = require('luatest.justrun')
+
+local g = t.group()
+
+-- Access to box.cfg from init script.
+g.test_access_box_cfg_init_script= function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_access_box_cfg_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+box.cfg({memtx_memory = 107374182})
+print('box.cfg.memtx_memory ' .. tostring(box.cfg.memtx_memory))
+os.exit()
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, 'box.cfg.memtx_memory 107374182')
+end
+
+-- The testcases tests insert tuples using fiber in init script.
+g.test_insert_tuples_fiber_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_fiber_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+local fiber = require('fiber')
+local json = require('json')
+
+box.cfg()
+local space = box.schema.space.create('tweedledum')
+space:create_index('primary', { type = 'hash' })
+
+function do_insert()
+    space:insert({1, 2, 4, 8})
+end
+
+fiber.create(do_insert)
+
+print(json.encode(space:select(nil, { limit = 4 })))
+
+os.exit()
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.stdout, '[[1,2,4,8]]')
+    t.assert_equals(res.exit_code, 0)
+end
+
+-- The testcase tests insert from detached fiber.
+g.test_insert_tuples_and_select_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_and_select_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+local yaml = require("yaml")
+local fiber = require("fiber")
+
+box.cfg()
+local space = box.schema.space.create("tweedledum")
+space:create_index("primary", { type = "hash" })
+
+fiber.create(function()
+    space:insert({1, 2, 4, 8})
+end)
+
+print(yaml.encode(space:select()))
+
+os.exit(0)
+    ]])
+    local opts = {nojson = true, stderr = true}
+    local res = justrun.tarantool(dir, {}, { script_name }, opts)
+    t.assert_equals(res.exit_code, 0)
+    t.assert_equals(res.stdout, [[
+---
+- [1, 2, 4, 8]
+...]])
+end
+
+-- The testcase tests insert from an init script.
+g.test_insert_tuples_init_script = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'test_insert_tuples_and_get_init_script.lua'
+    treegen.write_file(dir, script_name, [[
+box.cfg()
+local space = box.schema.space.create("tweedledum")
+space:create_index("primary", { type = "hash" })
+
+space:insert({2, 4, 8, 16})
+print(space:get(1))
```

Comment:
> Also, there was an additional insert "to avoid race conditions under valgrind",

Seems it is already removed.

Updated init scripts:

```diff
diff --git a/test/app-luatest/init_script_test.lua b/test/app-luatest/init_script_test.lua
index 5c13a9c847..a380c206b7 100644
--- a/test/app-luatest/init_script_test.lua
+++ b/test/app-luatest/init_script_test.lua
@@ -12,13 +12,13 @@ g.test_access_box_cfg_init_script= function()
     local script_name = 'test_access_box_cfg_init_script.lua'
     treegen.write_file(dir, script_name, [[
 box.cfg({memtx_memory = 107374182})
-print('box.cfg.memtx_memory ' .. tostring(box.cfg.memtx_memory))
+assert(box.cfg.memtx_memory == 107374182)
 os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0, 'exit code')
-    t.assert_equals(res.stdout, 'box.cfg.memtx_memory 107374182', 'stdout')
+    t.assert_equals(res.stdout, '', 'stdout')
 end
 
 -- The testcases tests insert tuples using fiber in init script.
@@ -27,7 +27,7 @@ g.test_insert_tuples_fiber_init_script = function()
     local script_name = 'test_insert_tuples_fiber_init_script.lua'
     treegen.write_file(dir, script_name, [[
 local fiber = require('fiber')
-local yaml = require('yaml')
+local json = require('json')
 
 box.cfg()
 local space = box.schema.space.create('tweedledum')
@@ -39,16 +39,14 @@ end
 
 fiber.create(do_insert)
 
-print(yaml.encode(space:select(nil, { limit = 4 })))
+assert(json.encode(space:select(nil, { limit = 4 })[1] == '[1,2,4,8]'))
 
 os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
-    t.assert_equals(res.stdout, '---\
-- [1, 2, 4, 8]\
-...', 'stdout')
     t.assert_equals(res.exit_code, 0, 'exit code')
+    t.assert_equals(res.stdout, '', 'stdout')
 end
 
 -- The testcase tests insert from detached fiber.
@@ -56,7 +54,7 @@ g.test_insert_tuples_and_select_init_script = function()
     local dir = treegen.prepare_directory({}, {})
     local script_name = 'test_insert_tuples_and_select_init_script.lua'
     treegen.write_file(dir, script_name, [[
-local yaml = require('yaml')
+local json = require('json')
 local fiber = require('fiber')
 
 box.cfg()
@@ -67,17 +65,15 @@ fiber.create(function()
     space:insert({1, 2, 4, 8})
 end)
 
-print(yaml.encode(space:select()))
+local str = json.encode(space:select(nil, { limit = 4 })[1])
+assert(str == '[1,2,4,8]')
 
 os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0, 'exit code')
-    t.assert_equals(res.stdout, [[
----
-- [1, 2, 4, 8]
-...]], 'stdout')
+    t.assert_equals(res.stdout, '', 'stdout')
 end
 
 -- The testcase tests insert from an init script.
@@ -85,25 +81,24 @@ g.test_insert_tuples_init_script = function()
     local dir = treegen.prepare_directory({}, {})
     local script_name = 'test_insert_tuples_and_get_init_script.lua'
     treegen.write_file(dir, script_name, [[
+local json = require('json')
+
 box.cfg()
 local space = box.schema.space.create('tweedledum')
 space:create_index('primary', { type = 'hash' })
 
 space:insert({2, 4, 8, 16})
-print(space:get(1) or 'nil')
-print(space:get(2) or 'nil')
-print(space:get(3) or 'nil')
-print(space:get(4) or 'nil')
+assert(space:get(1) == nil)
+assert(json.encode(space:get(2)) == '[2,4,8,16]')
+assert(space:get(3) == nil)
+assert(space:get(4) == nil)
 
 os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0, 'exit code')
-    t.assert_equals(res.stdout, 'nil\
-[2, 4, 8, 16]\
-nil\
-nil', 'stdout')
+    t.assert_equals(res.stdout, '', 'stdout')
 end
 
 -- The testcase checks that `math.floor` is reachable in the init
@@ -113,17 +108,15 @@ g.test_require_math_floor_init_script = function()
     local script_name = 'test_require_math_floor_init_script.lua'
     treegen.write_file(dir, script_name, [[
 local floor = math.floor
-print(floor(0.9))
-print(floor(1.1))
+assert(floor(0.9) == 0)
+assert(floor(1.1) == 1)
 
 os.exit(0)
     ]])
     local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0, 'exit code')
-    t.assert_equals(res.stdout, [[
-0
-1]], 'stdout')
+    t.assert_equals(res.stdout, '', 'stdout')
 end
 
 -- A test case for tarantool/tarantool#53.
```

- `yaml` was replaced with `json` for convenience
- `print` removed, added `assert()` to init scripts. Additionally, testcase checks stdout and exit code.